v1/src/lib.rs

```diff
@@ -906,9 +906,17 @@ fn key_exprs_to_influx_regex(path_exprs: &[&keyexpr]) -> String {
 }
 
 fn clauses_from_parameters(p: &str) -> ZResult<String> {
-    let time_range = TimeRange::from_str(p);
+    let parameters = Parameters::from(p);
     let mut result = String::with_capacity(256);
     result.push_str("WHERE kind!='DEL'");
+
+    let tr = parameters.time_range();
+    if tr.is_none() {
+        result.push_str(" ORDER BY time DESC LIMIT 1");
+        return Ok(result);
+    }
+
+    let time_range = tr.unwrap();
```

Comment:
```suggestion
    let time_range = match parameters.time_range() {
        Some(time_range) => time_range,
        None => {
            result.push_str(" ORDER BY time DESC LIMIT 1");
            return Ok(result);
        }
    };
```

Nitpick: removing an `unwrap` always feels reassuring ;-)